pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +85,132 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then check when status was changed delete it if the requested duration has passed
+	completeCond, found := condition.FindConditionByType(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete)
+	if found && completeCond.Status == corev1.ConditionTrue {
+		deleted, requeueAfter, err := r.checkTransitionTimeAndDelete(reqLogger, changeTierRequest, completeCond)
+		if deleted || err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: requeueAfter,
+		}, nil
+	}
+
+	err = r.changeTier(reqLogger, changeTierRequest, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Change of the tier is completed")
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: r.config.GetDurationBeforeChangeRequestDeletion(),
+	}, r.setStatusChangeComplete(changeTierRequest)
+}
+
+func (r *ReconcileChangeTierRequest) checkTransitionTimeAndDelete(log logr.Logger, changeTierRequest *toolchainv1alpha1.ChangeTierRequest, completeCond toolchainv1alpha1.Condition) (bool, time.Duration, error) {
+	log.Info("the ChangeTierRequest is completed so we can ignore it")
+	timeFromLastChange := time.Since(completeCond.LastTransitionTime.Time)
```

Comment:
I would rename this variable to `timeSinceCompletion` instead, but it's a minor suggestion.

Reply:
done 1b63eac